ml-agents/mlagents/trainers/trainer/rl_trainer.py

```diff
@@ -96,14 +94,14 @@ def _update_end_episode_stats(self, agent_id: str, optimizer: Optimizer) -> None
                 self.reward_buffer.appendleft(rewards.get(agent_id, 0))
                 rewards[agent_id] = 0
             else:
-                if isinstance(optimizer.reward_signals[name], RewardSignal):
+                if isinstance(optimizer.reward_signals[name], BaseRewardProvider):
                     self.stats_reporter.add_stat(
-                        optimizer.reward_signals[name].stat_name,
+                        f"Policy/{optimizer.reward_signals[name].name.capitalize()} Reward",
```

Comment:
Move this to a field/property on BaseRewardProvider?

Reply:
I think this is cleaner. @vincentpierre what do you think about this? (probably a separate PR).